### ISSUE TITLE
requests expecting content in response should error if they get a 204 status code

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/dor-services-client#83]
For example:

dor-services-app can currently return a 204 for refresh_metadata:

Logs on dor-services-stage,:

```
I, [2019-08-26T13:21:30.769614 #85943]  INFO -- : [64314fd4-8949-43f0-a15e-71ed312fd98e] Started POST "/v1/objects/druid:bb151dz4290/refresh_metadata" for 171.67.23.30 at 2019-08-26 13:21:30 -0700
I, [2019-08-26T13:21:30.770224 #85943]  INFO -- : [64314fd4-8949-43f0-a15e-71ed312fd98e] Processing by MetadataRefreshController#refresh as */*
I, [2019-08-26T13:21:30.770265 #85943]  INFO -- : [64314fd4-8949-43f0-a15e-71ed312fd98e]   Parameters: {"id"=>"druid:bb151dz4290"}
I, [2019-08-26T13:21:30.838717 #85943]  INFO -- : [64314fd4-8949-43f0-a15e-71ed312fd98e] Completed 204 No Content in 68ms
```

204 is NOT an okay response code for a refresh_metadata request.  So the dor-services-client should error in this case.  

Are there other cases where no content returned from dor-services-app should error?

**Comment by jcoyne** [comment]
@ndushay I'm pretty sure this is designed to return no response.

**Comment by ndushay** [comment]
So refresh_metadata updates the descMetadataDS with results from symphony.   Again, per info from Ben, apparently this step sometimes doesn't successfully update the descMetadata ... do we surface non-success in the HTTP response?

**Comment by ndushay** [comment]
closing as it should be addressed by sul-dlss/dor-services-app#350
